### PR TITLE
ramips: convert usb power to regulators

### PR DESCRIPTION
**Pull request by neheb**
These things are regulators. Should silence dmesg messages about using dummy regulators.

ping @DragonBluep

### CODE REVIEW

File: target/linux/ramips/dts/mt7621_d-team_pbr-m1.dts [hunk 1]
```diff
@@ -87,6 +74,33 @@
...
+	};
+
+};
+
+&xhci_ehci_port1 {
+	vbus-supply = <&reg_power_usb3>;
+};
+
+&ehci_port2 {
+	vbus-supply = <&reg_power_usb2>;
```
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_d-team_pbr-m1.dts:103]: Please move these vbus-supply properties into xhci node. ref:
https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/usb/mediatek%2Cmtk-xhci.yaml

The two regulators can be merged into one by `gpios = <&gpio 22 GPIO_ACTIVE_HIGH>, <&gpio 25 GPIO_ACTIVE_HIGH>;`

**Comment by DragonBluep** [comment]
> Done. I do wonder how mt7620 and mt7628 would be handled. They have separate ehci and ohci nodes.

The generic ehci and ohci driver doesn't support power regulator. Maybe we can add an userspace consumer instead. see: https://github.com/openwrt/openwrt/pull/16476.

**Comment by neheb** [comment]
Oh. That's unfortunate.

Btw can you check my LOW changes? something seems off...

**Comment by DragonBluep** [comment]
> Btw can you check my LOW changes? something seems off...

> The two regulators can be merged into one by `gpios = <&gpio 22 GPIO_ACTIVE_HIGH>, <&gpio 25 GPIO_ACTIVE_HIGH>;`

Sorry, I looked at the fixed-regulator source code and it seems that the `gpios` property can only handle one GPIO.

**Comment by neheb** [comment]
@DragonBluep removed all but mt7621 and ones that use multiple GPIOs.

File: target/linux/ramips/dts/mt7621_zbtlink_zbt-wg1602-v04.dtsi [hunk 2]
```diff
@@ -72,15 +80,13 @@
 			gpio-export,output = <1>;
 			gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
 		};
-
-		ext-usb {
-			gpio-export,name = "ext-usb";
-			gpio-export,output = <1>;
```
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_zbtlink_zbt-wg1602-v04.dtsi:78]: We need to use `enable-active-high` and `GPIO_ACTIVE_HIGH` here. I did some tests about this:
```
	gpio-export,output	GPIO_ACTIVE_	output level
		0		LOW		low
		1		LOW		high
		0		HIGH		low
		1		HIGH		high
```

Comment (neheb) [reply target/linux/ramips/dts/mt7621_zbtlink_zbt-wg1602-v04.dtsi:78]: Fixed.

**Review by DragonBluep — APPROVED** [review]

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Comment by neheb** [comment]
@DragonBluep for the remaining mt7621 devices, would vbus33-supply make sense?

**Comment by DragonBluep** [comment]
> @DragonBluep for the remaining mt7621 devices, would vusb33-supply make sense?

I suggest keeping it as it is. Though vusb33-supply should be working.

**Comment by mkrle** [comment]
It seems this has [broken](https://forum.openwrt.org/t/no-internet-after-openwrt-update-lte5398-m904/214772/17) USB on Zyxel LTE5398-M904. I don't have the device available to test, but I assume that the GPIO is not being set to high for some reason. Perhaps we need `regulator-boot-on`? Any ideas?

**Comment by neheb** [comment]
I just looked through the code to see what regulator-boot-on does

```
  4         /*
  3          * The signal will be inverted by the GPIO core if flagged so in the
  2          * descriptor.
  1          */
299         if (config->enabled_at_boot)
  1                 gflags = GPIOD_OUT_HIGH;
  2         else
  3                 gflags = GPIOD_OUT_LOW;
  4 

```

That's just weird. Probably should add it just in case.

**Comment by mkrle** [comment]
Just wanted to leave [this](https://forum.openwrt.org/t/openwrt-support-for-zyxel-lte5398-m904/140157/224?u=mkrlel) here as well in case the discussion ever comes up whether a GPIO controlling one of the USB ports is a regulator or not for lte5398-m904.

**Pull request merged**